brozzler/metrics.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional
+
+try:
+    from http_sd_registry.client import (
+        Client,
+        Env,
+        Registration,
+        Scheme,
+        format_self_target,
+    )
+    from http_sd_registry.config import ClientConfig
+except ImportError:
+    http_sd_registry = None
```

Comment:
Why would there be an `ImportError`? Is it because FOSS users may not have access to non-FOSS `http_sd_registry`? Consider explaining the rationale (regardless of what it is) as a comment here.